### DISCUSSION TITLE
Block Domain before rewrite

### DISCUSSION THREAD

**Discussion by Penholder7412** [AdguardTeam/AdGuardHome#7282] [Q&A] [answered]
Hello,
i have a simple question, in my environment i have a rewrite for my local services. `*.domain.com`
Now i want to block one of these services by adguard. `service.domain.com`

The problem is before the block it gets rewritten and therefore not blocked.
Even with an important it does not get blocked.
Is there someway to change the order to block before rewrite?

**Comment by schzhn** [comment]
Hello, @Penholder7412. For DNS rewrite use a custom filtering rule:
```
! block
||service.domain.com^

! rewrite
|*.domain.com^$dnsrewrite=NOERROR;CNAME;example.com
```

**Reply by Penholder7412** [reply]
Sadly this did not fix my issue.
I still get the dnsrewrite without the block.
Does it matter that i have it on a different port?
Currently the side to block is on the port 9443.

My current Config to rewrite to a local IP.
```
||service1.domain.com^$important
||service2.domain.com^$important

! rewrite
|*.domain.com^$dnsrewrite=NOERROR;A;10.0.0.100
```

**Reply by Penholder7412** [reply, marked as answer]
After tinkering with the syntax i found the solution.
All i was missing was the denyallow, after this it works like a charm.
Thank you for pointing me in the right direction.

```
||service1.domain.com^
||service2.domain.com^

|*.domain.com^$dnsrewrite=NOERROR;A;10.0.0.100,denyallow=service2.domain.com|service1.domain.com
```
